stacker/blueprints/base.py

```diff
@@ -246,14 +252,16 @@ def parse_user_data(variables, raw_user_data, blueprint_name):
         else:
             variable_values[key] = variables[key]
 
-    template = StringTemplate(raw_user_data)
+    template = string.Template(raw_user_data)
 
     res = ""
 
     try:
         res = template.substitute(variable_values)
-    except Exception as e:
-        raise MissingVariable(blueprint_name, e)
+    except ValueError:
```

Comment:
This gets raised when the placeholder name for the variable in the template is actually invalid. E.g ${100} raises a ValueError because the placeholder name (100) can not only be a number.